crates/store/re_rrdp_comms/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+//! Communications with an RRDP GRPC server.
+
+mod address;
+
+pub use address::{Address, InvalidAddressError};
+
+// ----------------------------------------------------------------------------
+
+use std::{error::Error, str::FromStr};
+
+use re_chunk::Chunk;
+use re_log_types::{
+    ApplicationId, LogMsg, SetStoreInfo, StoreId, StoreInfo, StoreKind, StoreSource, Time,
+};
+use re_remote_store_types::{
+    codec::{decode, CodecError},
+    v0::{
+        storage_node_client::StorageNodeClient, EncoderVersion, FetchRecordingRequest, RecordingId,
+    },
+};
+
+// ----------------------------------------------------------------------------
+
+/// Wrapper with a nicer error message
+#[derive(Debug)]
+struct TonicStatusError(tonic::Status);
+
+impl std::fmt::Display for TonicStatusError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let status = &self.0;
+        write!(f, "gRPC error, status: '{}'", status.code())?;
+        if !status.message().is_empty() {
+            write!(f, ", message: {:?}", status.message())?;
+        }
+        if !status.details().is_empty() {
+            write!(f, ", details: {:?}", status.details())?;
```

Comment:
small note - details is in binary form, so this wouldn't be human readable before we decode it (``prost::Message::decode``) and we have per-endpoint details error (protobuf defined error like ``RegistrationError``) so not sure what's the best way to deal with this.

Reply:
🤔  Yeah, maybe just skip it? In the few live tests I did, this was always empty

Reply:
skipped

Reply:
yeah, we only include it now for ``RegistrationError``, but (again, with the hopes we're aligned on structured error handling) we should have it for all endpoints and all calls. We can chose to expose that when we have it in place for more informative error messages and/or when client needs to actually decide what to do when error with error code _x_ occurrs.